nakama/socket.lua

```diff
@@ -0,0 +1,735 @@
+-- @param min_count
+-- @param max_count
+-- @param query
+-- @param string_properties
+-- @param numeric_properties
+-- @param count_multiple
+-- @param callback
+function M.party_matchmaker_add(socket, party_id, min_count, max_count, query, string_properties, numeric_properties, count_multiple, callback)
+	assert(socket)
+	assert(_G.type(party_id) == 'string')
+	assert(_G.type(min_count) == 'number')
+	assert(_G.type(max_count) == 'number')
+	assert(_G.type(query) == 'string')
+	assert(_G.type(string_properties) == 'table')
+	assert(_G.type(numeric_properties) == 'table')
+	assert(_G.type(count_multiple) == 'number')
+	local message = {
+		party_matchmaker_add = {
+			party_id = party_id,
+			min_count = min_count,
+			max_count = max_count,
+			query = query,
+			string_properties = string_properties,
+			numeric_properties = numeric_properties,
+			count_multiple = count_multiple,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+--- party_matchmaker_remove
+-- @param socket
+-- @param party_id
+-- @param ticket
+-- @param callback
+function M.party_matchmaker_remove(socket, party_id, ticket, callback)
+	assert(socket)
+	assert(_G.type(party_id) == 'string')
+	assert(_G.type(ticket) == 'string')
+	local message = {
+		party_matchmaker_remove = {
+			party_id = party_id,
+			ticket = ticket,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+--- party_data_send
+-- @param socket
+-- @param party_id
+-- @param op_code
+-- @param data
+-- @param callback
+function M.party_data_send(socket, party_id, op_code, data, callback)
+	assert(socket)
+	assert(_G.type(party_id) == 'string')
+	assert(_G.type(op_code) == 'number')
+	assert(_G.type(data) == 'string')
+	local message = {
+		party_data_send = {
+			party_id = party_id,
+			op_code = op_code,
+			data = data,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+--- status_follow
+-- @param socket
+-- @param user_ids
+-- @param usernames
+-- @param callback
+function M.status_follow(socket, user_ids, usernames, callback)
+	assert(socket)
+	assert(_G.type(user_ids) == 'string')
+	assert(_G.type(usernames) == 'string')
+	local message = {
+		status_follow = {
+			user_ids = user_ids,
+			usernames = usernames,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+--- status_unfollow
+-- @param socket
+-- @param user_ids
+-- @param callback
+function M.status_unfollow(socket, user_ids, callback)
+	assert(socket)
+	assert(_G.type(user_ids) == 'string')
+	local message = {
+		status_unfollow = {
+			user_ids = user_ids,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+--- status_update
+-- @param socket
+-- @param status
+-- @param callback
+function M.status_update(socket, status, callback)
+	assert(socket)
+	assert(_G.type(status) == 'string')
+	local message = {
+		status_update = {
+			status = status,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+
+
+--
+-- events
+--
+on_channel_presence_event
```

Comment:
Should this be commented out?

Reply:
🤦  Doh, I tweaked the template and regenerated without doing a final check. "I only did a small change so it should work right?!"

Sorry about that.